src/drivers/git.ts

```diff
@@ -0,0 +1,84 @@
+import fs, { existsSync, promises as fsp } from 'fs'
+import git from 'isomorphic-git'
+import http from 'isomorphic-git/http/node'
+import { defineDriver } from '..'
+import { join } from 'path'
+import { readdirRecursive, rmRecursive, writeFile, readFile, unlink } from './utils/node-fs'
+
+interface DriverOptions {
+  url?: string
+  ref?: string
+  path?: string
+}
+
+export default defineDriver((opts: DriverOptions = {}) => {
+  const gitOpts = {
+    fs,
+    http,
+    url: opts.url!,
+    dir: '/tmp/' + opts.url!.replace(/[^a-z0-9]/gi, '-'),
+    ref: opts.ref,
+    singleBranch: true,
+    noTags: true,
+    depth: 1
+  }
+  let initiated = false
+  const clone = async () => git.clone(gitOpts)
+  
+  const resolve = (...path: string[]) => join(gitOpts.dir, ...path)
+  
+  const r = (key: string) => resolve(opts.path || '/', key.replace(/:/g, '/'))
+
+  return {
+    
+    async hasItem (key) {
+      if (!initiated) {
```

Comment:
Refactor: We can create a utility like `ensureInitialized()` and call in several utils. Also caching one promise to avoid cloning multiple times in a race condition